src/metadata/trackmetadatataglib.cpp

```diff
@@ -0,0 +1,688 @@
+    }
+
+    const TagLib::ID3v2::FrameList keyFrame(tag.frameListMap()["TKEY"]);
+    if (!keyFrame.isEmpty()) {
+        pTrackMetadata->setKey(toQStringFirst(keyFrame));
+    }
+
+    // Foobar2000-style ID3v2.3.0 tags
+    // TODO: Check if everything is ok.
+    TagLib::ID3v2::FrameList textFrames(tag.frameListMap()["TXXX"]);
+    for (TagLib::ID3v2::FrameList::ConstIterator it(textFrames.begin());
+            it != textFrames.end(); ++it) {
+        TagLib::ID3v2::UserTextIdentificationFrame* replaygainFrame =
+                dynamic_cast<TagLib::ID3v2::UserTextIdentificationFrame*>(*it);
+        if (replaygainFrame && replaygainFrame->fieldList().size() >= 2) {
+            const QString desc(
+                    toQString(replaygainFrame->description()).toLower());
+            if (desc == "replaygain_album_gain") {
+                pTrackMetadata->setReplayGainDbString(
+                        toQString(replaygainFrame->fieldList()[1]));
+            }
+            // Prefer track gain over album gain.
+            if (desc == "replaygain_track_gain") {
+                pTrackMetadata->setReplayGainDbString(
+                        toQString(replaygainFrame->fieldList()[1]));
+            }
+        }
+    }
+
+    const TagLib::ID3v2::FrameList albumArtistFrame(tag.frameListMap()["TPE2"]);
+    if (!albumArtistFrame.isEmpty()) {
+        pTrackMetadata->setAlbumArtist(toQStringFirst(albumArtistFrame));
+    }
+
+    if (pTrackMetadata->getAlbum().isEmpty()) {
+        const TagLib::ID3v2::FrameList originalAlbumFrame(
+                tag.frameListMap()["TOAL"]);
+        pTrackMetadata->setAlbum(toQStringFirst(originalAlbumFrame));
+    }
+
+    const TagLib::ID3v2::FrameList composerFrame(tag.frameListMap()["TCOM"]);
+    if (!composerFrame.isEmpty()) {
+        pTrackMetadata->setComposer(toQStringFirst(composerFrame));
+    }
+
+    const TagLib::ID3v2::FrameList groupingFrame(tag.frameListMap()["TIT1"]);
+    if (!groupingFrame.isEmpty()) {
+        pTrackMetadata->setGrouping(toQStringFirst(groupingFrame));
+    }
+
+    // ID3v2.4.0: TDRC replaces TYER + TDAT
+    const TagLib::ID3v2::FrameList recordingDateFrame(
+            tag.frameListMap()["TDRC"]);
+    if (!recordingDateFrame.isEmpty()) {
+        pTrackMetadata->setYear(toQStringFirst(recordingDateFrame));
+    }
+}
+
+void readAPETag(TrackMetadata* pTrackMetadata, const TagLib::APE::Tag& tag) {
+    if (kDebugMetadata) {
+        for (TagLib::APE::ItemListMap::ConstIterator it(
+                tag.itemListMap().begin()); it != tag.itemListMap().end();
+                ++it) {
+            qDebug() << "APE" << toQString((*it).first) << "-"
+                    << toQString((*it).second.toString());
+        }
+    }
+
+    readTag(pTrackMetadata, tag);
+
+    if (tag.itemListMap().contains("BPM")) {
+        pTrackMetadata->setBpmString(toQString(tag.itemListMap()["BPM"]));
+    }
+
+    if (tag.itemListMap().contains("REPLAYGAIN_ALBUM_GAIN")) {
```

Comment:
On a second thought: I think Mixxx should not use the album gain at all. This is not the Gain we want, reanalyze will be the best solution for this case.